indra/newview/llagentiomodule.h

```diff
@@ -0,0 +1,94 @@
+/** 
+ * @file llagentiomodule.h
+ * @brief Implementation of llagentiomodule class.
+ *
+ * $LicenseInfo:firstyear=2001&license=viewerlgpl$
+ * Second Life Viewer Source Code
+ * Copyright (C) 2022, Linden Research, Inc.
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License only.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * 
+ * Linden Research, Inc., 945 Battery Street, San Francisco, CA  94111  USA
+ * $/LicenseInfo$
+ */
+
+#pragma once
+#ifndef LL_LLAGENTMODULE_H
+#define LL_LLAGENTMODULE_H
+
+//-----------------------------------------------------------------------------
+// Header files
+//-----------------------------------------------------------------------------
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "llevents.h"
+#include "lleventapi.h"
+#include "lazyeventapi.h"
+#include "llsingleton.h"
+
+class LLLeap;
+using namespace LL;
+
+class LLAgentIOModule : public LLSingleton<LLAgentIOModule>,
+	public LLEventAPI
```

Comment:
n.b. `LLEventAPI` already isa `LLInstanceTracker` by its name, so it doesn't need to be an `LLSingleton` to find it.